components/mdc/TextInput/MoneyInput.svelte

```diff
@@ -31,7 +32,9 @@ $: isLowerThanMinValue = minValue && internalValue < minValue
 $: showErrorIcon = hasExceededMaxValue || isLowerThanMinValue || hasExceededMaxLength || valueNotDivisibleByStep
 $: error = showErrorIcon || (hasFocused && hasBlurred && required && !internalValue)
 $: showCounter = maxlength && valueLength / maxlength > 0.85
-$: valueNotDivisibleByStep = internalValue && (internalValue / Number(step)).toFixed(2) % 1 !== 0
+$: valueHasTooManyDecPlaces = getDecimalPlacesLength(internalValue) > getDecimalPlacesLength(step)
+$: valueNotDivisibleByStep =
+  (internalValue && (internalValue / Number(step)).toFixed(2) % 1 !== 0) || valueHasTooManyDecPlaces
```

Comment:
I don't really understand the logic here, which suggests that it may be worth extracting and/or simplifying something about it, but if it works for the use cases you've tested and you believe it's an improvement over the current version, we can go ahead and proceed with this change.

Reply:
When a person types in 1.0001 when the step is .001 instead of checking for a remainder, `valueHasTooManyDecPlaces` check to see if .0001's length is longer than .001. 
If .0000 is entered it resolved to 0 so it passes the check. Maybe renaming valueHasTooManyDecPlaces to something like tooManyDecPlacesToBeDivisible would be better?

Reply:
No, I think your `valueHasTooManyDecPlaces` name is good. No need to change that one. I guess it's primarily the `...toFixed(2)` that's confusing me. Why 2? Should that be the number of decimal places in step?

Reply:
2 is just the number of decimal places I have found works well for eliminating the problems associated with using floats in fractions while still being sensitive to long decimals. See [this](https://stackoverflow.com/questions/588004/is-floating-point-math-broken) thread. I have tried various things, but 2 works the best. There may be a better solution, but as @Schparky said, that might involve rewriting it in Go,